File: revisão/python/exerc_05.py
def main():
    try:
        a, b = map(int, input("Digite dois inteiros (a b): ").split())
    except Exception:
        print("Entrada inválida.")
        return

    # Aritmética
    soma = a + b
    sub = a - b
    mul = a * b
    div_int = a // b if b != 0 else None
    div_real = a / b if b != 0 else None
    resto = a % b if b != 0 else None
    media = (a + b) / 2.0

    print("\n--- Aritmética ---")
    print(f"soma = {soma}\nsub = {sub}\nmul = {mul}\ndiv int = {div_int}\ndiv real = {div_real}\nresto = {resto}\nmedia = {media:.2f}")

    # Relacionais
    print("\n--- Relacionais ---")
    print("a > b ?", a > b)
    print("a == b ?", a == b)
    print("Ambos pares?", (a % 2 == 0 and b % 2 == 0))

    # Decisão
    print("\n--- Decisão ---")
    if b == 0:
        print("b é zero (cuidado).")
    elif media >= 10.0:
        print("Média >= 10.0")
    else:
        print("Média < 10.0")

    # Repetição
    print("\n--- Repetições ---")
    print(f"Tabela de {a} (for 1..5):")
    for i in range(1, 6):
        print(f"{a} x {i} = {a * i}")

    temp = b
    print("\nContagem regressiva (while):")
    if temp <= 0:
        print("(b <= 0, não entrou no while)")
    else:
        while temp > 0:
            print(temp, end=' ')
            temp -= 1
        print()

    # Python não tem do/while — simular
    print("\nSimulação do/while (executa pelo menos 1 vez):")
    temp = 2
    while True:
        print("temp =", temp)
        temp -= 1
        if not (temp > 0):
            break

if __name__ == "__main__":
    main()

File: revisão/java/exerc_05.java
import java.util.Scanner;

public class Demo {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Digite dois inteiros (a b): ");
        if (!sc.hasNextInt()) { System.out.println("Entrada inválida."); return; }
        int a = sc.nextInt();
        if (!sc.hasNextInt()) { System.out.println("Entrada inválida."); return; }
        int b = sc.nextInt();

        int soma = a + b;
        int sub = a - b;
        int mul = a * b;
        int div_int = (b != 0) ? (a / b) : 0;
        double div_real = (b != 0) ? (double)a / b : 0.0;
        int resto = (b != 0) ? a % b : 0;
        double media = (a + b) / 2.0;

        System.out.println("\n--- Aritmética ---");
        System.out.printf("soma = %d\nsub = %d\nmul = %d\ndiv int = %d\ndiv real = %.4f\nresto = %d\nmedia = %.2f\n",
                          soma, sub, mul, div_int, div_real, resto, media);

        System.out.println("\n--- Relacionais ---");
        System.out.println("a > b ? " + (a > b));
        System.out.println("a == b ? " + (a == b));
        System.out.println("Ambos pares? " + (a % 2 == 0 && b % 2 == 0));

        System.out.println("\n--- Decisão ---");
        if (b == 0) System.out.println("b é zero.");
        else if (media >= 10.0) System.out.println("Média >= 10.0");
        else System.out.println("Média < 10.0");

        System.out.println("\n--- Repetições ---");
        System.out.println("Tabela de " + a + " (for 1..5):");
        for (int i = 1; i <= 5; i++) System.out.printf("%d x %d = %d\n", a, i, a * i);

        int temp = b;
        System.out.println("\nContagem regressiva (while):");
        while (temp > 0) {
            System.out.print(temp + " ");
            temp--;
        }
        if (b <= 0) System.out.println("(b <= 0, não entrou no while)");
        else System.out.println();

        temp = 2;
        System.out.println("\nExemplo do/while:");
        do {
            System.out.println("temp = " + temp);
            temp--;
        } while (temp > 0);

        sc.close();
    }
}